docs/source/en/main_classes/tokenizer.md

```diff
@@ -51,6 +51,25 @@ token space (e.g., getting the index of the token comprising a given character o
 to a given token).
 
 
+# Multimodal Tokenizer
+
+Apart from that each tokenizer can be a "multimodal" tokenizer which means that the tokenizer will hold all relevant special tokens
+as part of tokenizer attributes for easier access. For example, if the tokenizer is loaded from a vision-language model like LLaVA, you will
+be able to access `tokenizer.image_token_id` to obtain the special image token used as a placeholder. 
+
+To enable extra special tokens for any type of tokenizer, you have to add the following lines and save the tokenizer. Extra special tokens do not
+have to be modality related and can ne anything that the model often needs access to. In the below code, tokenizer at `output_dir` will have direct access
+to three more special tokens.  
+
+```python
+tokenizer = AutoTokenizer.from_pretrained(model_id)
+tokenizer.extra_special_tokens = ["image_token", "boi_token", "eoi_token"]
```

Comment:
```suggestion
tokenizer = AutoTokenizer.from_pretrained(model_id, extra_special_tokens = ["image_token", "boi_token", "eoi_token"])
```

Reply:
let's add a small test for this

Reply:
yes, this is actually not correct anymore hehe, forgot to update the docs. And it has a test for that already so we are good

new way of adding extra special tokens is like
`tokenizer.extra_special_tokens = {"eoi_token": "<s>", "image_token": "<image>"}`. After adding this line and saving the tokenizer, loading back will do the magic and tokenizer will have `self.image_token` attribute

Reply:
we should be able to pass it as input as well instead of forcing people to use the setter! 🤗

Reply:
yeap, realized later and added that in the docs instead of "saving-loading back". Plus extended the test